src/components/Button/Button.tsx

```diff
@@ -68,44 +69,45 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
       rightIcon && (
         <span
           className={cn(iconClassName, styles.icon, styles.right, {
-            [styles.noText]: !text || textIsHidden
+            [styles.noText]: noText
           })}
         >
           {rightIcon}
         </span>
       )
 
+    const getAriaLabel = () => {
```

Comment:
super DUPER extra nit-picky: maybe comment with the intent here, something like:
```ts
// Use the text prop as the aria-label if the text becomes hidden
// and no aria-label was provided to keep the button accessible.
```

Reply:
totally! its def a weird case